pkg/x/vmwscript/vm_provision.go

```diff
@@ -185,13 +185,43 @@ func RunTasks(ctx context.Context, client *govmomi.Client) {
 				Password: *vm.VMTemplateAuth.Password,
 			}
 
+			// Check if a networking configuration exists
+			if task.Task.Network != nil {
+				// Determine the distribution to configure the networking
+				if task.Task.Network.Distro == "centos" {
```

Comment:
How about a switch here:

```
switch task.Task.Network.Distro {
   case "centos" :  // call a function here.
   default :  // return some 'unsupported' error.
}
```

This makes it clear that the Distro field will be something important to add more cases in the future...